clang-tools-extra/clang-tidy/misc/AlphabeticalDeclOrderCheck.cpp

```diff
@@ -0,0 +1,84 @@
+//===--- AlphabeticalDeclOrderCheck.cpp - clang-tidy ----------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "AlphabeticalDeclOrderCheck.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+
+using namespace clang::ast_matchers;
+
+namespace clang {
+namespace tidy {
+namespace misc {
+
+void AlphabeticalDeclOrderCheck::registerMatchers(MatchFinder *Finder) {
+  Finder->addMatcher(declStmt().bind("stmt"), this);
+}
+
+bool AlphabeticalDeclOrderCheck::checkAlphabeticalOrder(const DeclStmt *MatchedStmt, std::vector<StringRef> &DeclNames, bool &fixDecl) {
+  auto prev_it = MatchedStmt->decl_begin();
+  auto it = MatchedStmt->decl_begin();
+  auto firstVarDecl = dyn_cast<VarDecl>(*MatchedStmt->decl_begin());
```

Comment:
I don't mind whether we do `firstVarDecl` or `FirstVarDecl` for variables since this isn't done consistently in LLVM. But whatever you choose, we should at least do it consistently in this file.